src/build.rs

```diff
@@ -258,28 +270,25 @@ impl BuildQueue {
             }
         }
 
-        let cmd_line_args = self.cmd_line_args.lock().unwrap();
-        assert!(!cmd_line_args.is_empty());
-        let cmd_line_envs = self.cmd_line_envs.lock().unwrap();
-        self.rustc(&*cmd_line_args, &*cmd_line_envs, build_dir)
+        let compile_cx = self.compilation_cx.lock().unwrap();
+        let (args, envs) = (&(*compile_cx).args, &(*compile_cx).envs);
```

Comment:
This line looks a bit weird - I would make it two `let`s without a tuple. I'm surprised you'd need the explicit `&*` too